### PR TITLE
Cumulus v18.3.1 Upgrade

### PR DESCRIPTION
**Pull request by gjclark**

### CODE REVIEW

**Comment by mattp0** [comment]
Curious what people think we should do here in core: 

CUMULUS-3433 Update to node.js v20
The following applies only to users with a custom value configured for
async_operation_image:

As part of the node v20 update process, a new version (49) of the Core
async-operation container was published - cumuluss/async
operation The
default value for async_operation_image has been updated in the cumulus
module, however if you are using an internal image repository such as ECR,
please make sure to update your deployment configuration with the newly
provided image.

Users making use of a custom image configuration should note the base image
for Core async operations must support node v20.x.

CUMULUS-3617 Migration of DLA messages should be performed after Cumulus is upgraded
Instructions for migrating old DLA (Dead Letter Archive) messages to new format:

YYYY-MM-DD subfolders to organize by date
new top level fields for simplified search and analysis
captured error message
To invoke the Lambda and start the DLA migration, you can use the AWS Console or CLI:

aws lambda invoke --function-name $PREFIX-migrationHelperAsyncOperation \
  --payload $(echo '{"operationType": "DLA Migration"}' | base64) $OUTFILE
PREFIX is your Cumulus deployment prefix.
OUTFILE (optional) is the filepath where the Lambda output will be saved.
The Lambda will trigger an Async Operation and return an id such as:

{"id":"41c9fbbf-a031-4dd8-91cc-8ec2d8b5e31a","description":"Migrate Dead Letter Archive Messages",
"operationType":"DLA Migration","status":"RUNNING",
"taskArn":"arn:aws:ecs:us-east-1:AWSID:task/$PREFIX-CumulusECSCluster/123456789"}
which you can then query the Async Operations API
Endpoint for the
output or status of your request. If you want to directly observe the progress
of the migration as it runs, you can view the CloudWatch logs for your async
operations (e.g. PREFIX-AsyncOperationEcsLogs).

CUMULUS-3779 async_operations Docker image version upgrade
The async-operation Docker image has been updated to support Node v20 and aws-sdk v3. Users of the image will need
to update to at least [async-operations:52](https://hub.docker.com/layers/cumuluss/async-operation/52/images/sha256-78c05f9809c29707f9da87c0fc380d39a71379669cbebd227378c8481eb11c3a?context=explore).

CUMULUS-3776 cumulus-ecs-task Docker image version upgrade
The cumulus-ecs-task Docker image has been updated to support Node v20 and aws-sdk v3. Users of the image will need
to update to at least [cumulus-ecs-task:2.1.0](https://hub.docker.com/layers/cumuluss/cumulus-ecs-task/2.1.0/images/sha256-17bebae3e55171c96272eeb533293b98e573be11dd5371310156b7c2564e691a?context=explore).


there appears to be nothing to be done in core for this. The lambda is an invoke with this payload. IMO its seems like it wont need a migration script. 

The container updates are more an FYI..

**Comment by mattp0** [comment]
Some context for why we are moving to Amazon Linux 2023. Node 20.x can not be installed on Amazon Linux 2 because the node system required a GLIBC 2.28 and Amazon Linux 2 does not have a non trivial way of doing this. We found many blog posts of people asking if GLIBC 2.27+ is possible on Amazon Linux 2 with no solutions. 

What this means is the baseline python version is now 3.9. ASF is prepared to accept this and handle updating our dependencies. We would like feedback from other DAACs here if this is reasonable. 

There is a chance we could install 3.8 on the container but if no one has a strong reason to support 3.8 I would be in favor of moving forward from 3.8 as its EOL in Oct 2024. 

What do you all think? @lindsleycj @mikedorfman

File: Dockerfile [hunk 1]
```diff
@@ -9,19 +9,16 @@ FROM amazonlinux:2 as core_base
...
 #   * Terraform
 
-ENV NODE_VERSION "16.x"
-ENV TERRAFORM_VERSION "1.5.3"
-ENV AWS_CLI_VERSION "2.13.25"
+ENV NODE_VERSION="20.x"
+ENV TERRAFORM_VERSION="1.9.2"
+ENV AWS_CLI_VERSION="2.17.13"
 
 # Add NodeJS and Yarn repos & update package index
```
Comment (mattp0) [inline Dockerfile:16]: This is incorrect, should update the comment

File: Dockerfile [hunk 2]
```diff
@@ -33,35 +30,24 @@ RUN \
...
-        yum install -y session-manager-plugin.rpm
+        dnf install -y session-manager-plugin.rpm
 
 # Add user for keygen in Makefile
 ARG USER
 RUN \
         echo "user:x:${USER}:0:root:/:/bin/bash" >> /etc/passwd
 
-COPY .gitconfig /.gitconfig
+#COPY .gitconfig /.gitconfig
```
Comment (mattp0) [inline Dockerfile:43]: Maybe we should add context to why this is here and why its an issue for ASF.

**Review by mattp0 — APPROVED** [review]

File: Dockerfile [hunk 3]
```diff
@@ -9,19 +9,16 @@ FROM amazonlinux:2 as core_base
...
+ENV TERRAFORM_VERSION="1.9.2"
+ENV AWS_CLI_VERSION="2.17.13"
 
 # Add NodeJS and Yarn repos & update package index
-RUN \
-        yum install https://rpm.nodesource.com/pub_${NODE_VERSION}/nodistro/repo/nodesource-release-nodistro-1.noarch.rpm -y && \
-        yum install nodejs -y --setopt=nodesource-nodejs.module_hotfixes=1 && \
-        curl -sL https://dl.yarnpkg.com/rpm/yarn.repo | tee /etc/yum.repos.d/yarn.repo && \
-        yum update -y
+RUN curl -fsSL https://rpm.nodesource.com/setup_20.x | bash -
```
Comment (mattp0) [inline Dockerfile:17]: Should return this to the variable method:
```suggestion
RUN curl -fsSL https://rpm.nodesource.com/setup_${NODE_VERSION} | bash -
```

Comment (gjclark) [reply Dockerfile:17]: Shoot. Didn't mean to leave that there.

Comment (gjclark) [reply Dockerfile:17]: Good catch.

File: Makefile [hunk 4]
```diff
@@ -19,7 +19,7 @@
 #  PYTHON_VER: 			  python3 or python38 which sets the build target in make file
```
Comment (mattp0) [inline Makefile:19]: I just noticed this is now incorrect. There is no 38 target now

**Review by mattp0 — APPROVED** [review]

File: cumulus/variables.tf [hunk 5]
```diff
@@ -438,7 +438,7 @@ variable "cmr_acl_based_credentials" {
   description = "Option to enable/disable user based CMR ACLs to derive permission for s3 credential access tokens"
 }
 
-variable "thottled_queue_execution_limit" {
+variable "throttled_queue_execution_limit" {
```
Comment (mckadesorensen) [inline cumulus/variables.tf:441]: nice catch

**Review by mckadesorensen — APPROVED** [review]

**Comment by lindsleycj** [comment]
@mattp0 

> There is a chance we could install 3.8 on the container but if no one has a strong reason to support 3.8 I would be in favor of moving forward from 3.8 as its EOL in Oct 2024.
> 
> What do you all think? @lindsleycj @mikedorfman

I am fine with removing python 3.8 from this.

**Review by lindsleycj — APPROVED** [review]
I'm fine with these changes.

**Review by mikedorfman — APPROVED** [review]

**Pull request merged**